datalad/support/gitrepo.py

```diff
@@ -886,17 +886,29 @@ def __init__(self, path, url=None, runner=None, create=True,
             self._check_git_version()
 
         # BEGIN Repo validity test
-        # We want to fail early for tests, that would be performed a lot. In particular this is about
-        # GitRepo.is_valid_repo. We would use the latter to decide whether or not to call GitRepo() only for
-        # __init__ to then test the same things again. If we fail early we can save the additional test from outer
-        # scope.
+        # We want to fail early for tests, that would be performed a lot. In
+        # particular this is about GitRepo.is_valid_repo. We would use the
+        # latter to decide whether or not to call GitRepo() only for __init__ to
+        # then test the same things again. If we fail early we can save the
+        # additional test from outer scope.
         self.path = path
 
-        # Note, that the following three path objects are used often and therefore
-        # are stored for performance. Path object creation comes with a cost. Most noteably,
-        # this is used for validity checking of the repository.
+        # Note, that the following three path objects are used often and
+        # therefore are stored for performance. Path object creation comes with
+        # a cost. Most notably, this is used for validity checking of the
+        # repository.
         self.pathobj = ut.Path(self.path)
         self.dot_git = self._get_dot_git(self.pathobj, ok_missing=True)
+
+        # We'll need some awareness of whether or not this is a bare repo, if we
+        # are to support it beyond instantiation
+        # TODO: We should prob. double check config for core.bare=true! However,
+        #       at this point it would sabotage the lazy loading of
+        #       ConfigManager. May be `bare` property, that would check config +
+        #       path == dot_git for consistency? Once at it: To what extend can
+        #       this (or the location of .git in general) change during runtime?
```

Comment:
Just noted, that someone merged the destruction of lazy loading `ConfigManager` already ... Hmpf.